x-pack/legacy/plugins/alerting/server/routes/update.ts

```diff
@@ -36,6 +37,9 @@ export const updateAlertRoute = {
           throttle: getDurationSchema()
             .required()
             .allow(null),
+          tags: Joi.array()
+            .items(Joi.string())
+            .required(),
```

Comment:
Why required?  I guess we make most of our stuff required, but this certainly seems like one of those things that could be optional, defaulting to `[]`.

Reply:
This goes along the AAD issue where we can't do partial updates on ESOs. So even if the value doesn't change, it has to be passed in to not break the AAD. The throttle attribute shares the same issue. We could probably improve on this in the future and grab the current value from the existing object if not passed in to the API. Created a card: https://github.com/elastic/kibana/projects/26#card-28297595.

Reply:
I think allowing it as optional on update the leaves the question of whether, on update, if no tags are provided at all, does that mean don't update them, or delete them all.  In the user's mind anyway.  So, perhaps the way you've got it now is the best.